timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosResourcesFactory.java

```diff
@@ -234,7 +240,8 @@ private static ImmutablePaxosResources.Builder setupTimestampResources(
                             combinedNetworkClientFactories.learner().create(client),
                             paxosRuntime.get().maximumWaitBeforeProposalMs()),
                     client);
-            return PersistentTimestampServiceImpl.create(boundStore);
+            Object clientLock = clientLocks.computeIfAbsent(client, ignored -> new Object());
+            return PersistentTimestampServiceImpl.create(boundStore, () -> clientLock);
```

Comment:
My bias for some reason pushed for a supplier here rather than just an object. Don't feel strongly about this, but figured it was a bit cleaner, as in the future this would allow us to introduce more complex locking strategies, if needed. 

Alternatively: We could just pass the `Client` down instead, and then have a `static` concurrent map to handle these locks in the implementing class; I found this to be a bit uglier.

Reply:
Any thoughts on using a `ReentrantLock` here? I do get that the object is sufficient since the acquisition is block-structured, though I think that communicates intent more explicitly than a `Supplier<Object>` where it's possible for the returned object to change.

Definitely agree with using the Map here though. In my mind the "nice" design is a `Map<Client, MutualExclusionClass>` of some kind declared here, and then passing down the `MutualExclusionClass` for the relevant client which handles that (though it could be a bit over-engineered).

Reply:
Should we use a [Guava `Striped<Lock>`](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/util/concurrent/Striped.html), e.g. `Striped<Lock> stripedLocks = Striped.lock(remoteClients.context().remoteUris().size());`?